### ISSUE TITLE
Rule proposal: sort-import

### ISSUE THREAD

**Issue by gajus** [benmosher/eslint-plugin-import#162]
A rule that requires alphabetical sorting of the import statements. Sorting should be done based on the import path, irrespective of the variables being imported, e.g.

```js
import 'aaa';
import 'bbb';
import 'ccc';
```

Related:

* https://github.com/benmosher/eslint-plugin-import/issues/163

**Comment by gajus** [comment]
Probably unnecessary given https://github.com/eslint/eslint/blob/master/docs/rules/sort-imports.md

**Comment by benmosher** [comment]
ah, neat. good to know. tough to keep up with what is in ESLint proper.

**Comment by gajus** [comment]
That implementation has problems though, e.g.

```js
import React, {
    PropTypes
} from 'react';
```

will complain that `PropTypes` must come before `React`.

**Comment by wyze** [comment]
That was a bug that is now fixed. I made my own rule as I didn't like the way they handle single vs. multiple imports.

Example:
```js
// valid
import { b, e } from './b'
import a from './a'
import { c } from './c'
import d from './d'
```

I found that a little unreadable and made my own so that the importing of named exports would be grouped together.

**Comment by benmosher** [comment]
FYI: https://www.npmjs.com/package/eslint-plugin-import-order exists, might be worth considering forwarding sort order rules to that plugin?

**Comment by gajus** [comment]
@benmosher Is there a reason for not merging your changes to this plugin? It is a lot more generic.

**Comment by benmosher** [comment]
@gajus - I'm not sure what you mean, but it looks like we'll be merging the one rule from that plugin into this one.

**Comment by gajus** [comment]
Thats what I have meant.
